src/gui/core/window_builder.hpp

```diff
@@ -100,17 +100,17 @@ register_builder_widget(const std::string& id, widget_builder_func_t functor);
 builder_widget_ptr create_builder_widget(const config& cfg);
 
 /**
- * Helper to generate a widget from a WML widget instance.
+ * Helper to create a generic widget builder with the given WML data.
  *
  * Mainly used as functor for @ref register_builder_widget.
  *
  * @param cfg                     The config with the information to
- *                                Instantiate the widget.
+ *                                instantiate the widget.
  *
  * @returns                       A generic widget builder pointer.
  */
-template <class T>
-builder_widget_ptr build_widget(const config& cfg)
+template<typename T>
+builder_widget_ptr make_builder(const config& cfg)
```

Comment:
What even is the point of this function if it just calls `std::make_shared()`? IMO, it would be clearer to simply call `std::make_shared()` directly.